File: OpenMP_cpu/adv_intro/1_env_apis/solutions/ans_env_apis.c
#include <stdio.h>
#include <omp.h>

int main(){

// TODO 3: display the "ENV" and affinity
     omp_display_env(1);         // does not work on Lumi cce 15.0
     omp_display_affinity(NULL);

// TODO 1: get the thread number and number of threads
   #pragma omp parallel
   {   
      printf( " %d out of %d\n", omp_get_thread_num(), omp_get_num_threads() );
   }   

}

File: OpenMP_cpu/adv_intro/1_env_apis/solutions/ans_env_apis.f90
program main
use omp_lib

!! TODO 3: display the "ENV" and affinity
   !! call omp_display_<TODO>;
   !! call omp_display_<TODO>;
      call omp_display_env(.TRUE.)   !!Lumi not supported in cce 15.0
      call omp_display_affinity("")

!! TODO 1: get the thread number and number of threads
   !$omp parallel
     !!print*," omp_get_<TODO>, " out of " omp_get_<TODO>
       print*, omp_get_thread_num(), " out of ", omp_get_num_threads()
   !$omp end parallel

end program main
